website/project/model.py

```diff
@@ -3580,6 +3629,17 @@ def _on_reject(self, user, token):
         )
 
 
+class DraftRegistrationApproval(Sanction):
+
+    mode = Sanction.ANY
+
+    def _on_complete(self, user, token):
+        pass  # draft approval state gets loaded dynamically from this record
```

Comment:
Excellent summary @samchrisinger. We haven't used much real abstraction throughout our classes leading to things like the infamous `Node` class. `Sanction` is the first such, well thought out, abstract set of classes that I've seen so far. It's simply unintuitive for us to look *up* the chain of classes to look for these things -- hopefully that will change as our developers continue to grow.